src/rgw/driver/rados/rgw_sync_error_repo.cc

```diff
@@ -118,16 +117,17 @@ int remove(librados::ObjectWriteOperation& op,
 }
 
 class RGWErrorRepoWriteCR : public RGWSimpleCoroutine {
-  RGWSI_RADOS::Obj obj;
+  librados::Rados* rados;
+  const rgw_raw_obj& raw_obj;
```

Comment:
@smanjara it's hard to tell which coroutine is responsible for the crash in https://github.com/ceph/ceph/pull/54278#issuecomment-1841439328, but `RGWErrorRepoWriteCR` and `RGWErrorRepoWriteCR` probably need a copy of this `rgw_raw_obj` rather than a reference

Reply:
opened https://github.com/ceph/ceph/pull/54841 to see if that helps

Reply:
oh thanks @cbodley . I tried making a copy of the obj and tested locally. it still crashes.

Reply:
the trace is even more obvious now:

```
 1: /home/smanjara/ceph/build/bin/radosgw(+0x16fb694) [0x55bad2950694]
 2: /lib64/libc.so.6(+0x3dbb0) [0x7f896145fbb0]
 3: (librados::v14_2_0::Rados::ioctx_create(char const*, librados::v14_2_0::IoCtx&)+0x19) [0x7f896367c1c5]
 4: (rgw_init_ioctx(DoutPrefixProvider const*, librados::v14_2_0::Rados*, rgw_pool const&, librados::v14_2_0::I
oCtx&, bool, bool, bool)+0x50) [0x55bad25ec802]
 5: (rgw_get_rados_ref(DoutPrefixProvider const*, librados::v14_2_0::Rados*, rgw_raw_obj, rgw_rados_ref*)+0x57)
 [0x55bad25ed2bc]
 6: (rgw::error_repo::RGWErrorRepoWriteCR::send_request(DoutPrefixProvider const*)+0xa7) [0x55bad2817873]

Reply:
we also have create ioctx returning r=`-107`. not sure what that means in librados context

Reply:
> we also have create ioctx returning r=`-107`. not sure what that means in librados context

ok that's a boost asio error ENOTCONN when creating ioctx for the log pool

Reply:
we get the error r=-107 'Transport endpoint not connected' from the call to ioctx_create() 
https://github.com/ceph/ceph/blob/main/src/rgw/driver/rados/rgw_tools.cc#L29
I still don't understand why. the rgw_get_rados_ref() call in RGWErrorRepoWriteCR looks correct to me. not sure what I'm missing.

Reply:
somehow we're passing a nullptr for the `rados` argument:
```
Thread 1264 "data-sync" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffd171b6c0 (LWP 169135)]
librados::v14_2_0::Rados::ioctx_create (this=this@entry=0x0, name=0x7fffd1717628 "a1.rgw.log", io=...)
    at /home/cbodley/ceph/src/librados/librados_cxx.cc:2635
2635      int ret = rados_ioctx_create((rados_t)client, name, &p);
(gdb) bt
#0  librados::v14_2_0::Rados::ioctx_create (this=this@entry=0x0, name=0x7fffd1717628 "a1.rgw.log", io=...)
    at /home/cbodley/ceph/src/librados/librados_cxx.cc:2635
#1  0x00005555562fee82 in rgw_init_ioctx (dpp=<optimized out>, rados=rados@entry=0x0, pool=..., ioctx=..., create=create@entry=true, 
    mostly_omap=mostly_omap@entry=false, bulk=false) at /home/cbodley/ceph/src/rgw/driver/rados/rgw_tools.cc:29
#2  0x00005555562ff93c in rgw_get_rados_ref (dpp=<optimized out>, dpp@entry=0x55556253a218, rados=0x0, obj=..., ref=ref@entry=0x7fffd1717610)
    at /home/cbodley/ceph/src/rgw/driver/rados/rgw_tools.cc:105
#3  0x000055555651dabb in rgw::error_repo::RGWErrorRepoWriteCR::send_request (this=0x55555c632300, dpp=0x55556253a218)
    at /home/cbodley/ceph/src/rgw/driver/rados/rgw_sync_error_repo.cc:141
```

Reply:
may be we should try to call get_rados_handle() within ErrorRepoCR?

Reply:
oh, the RGWErrorRepoWrite/RemoveCR constructors just aren't copying the `rados` constructor argument into their member variable

Reply:
i updated https://github.com/ceph/ceph/pull/54841 accordingly

Reply:
huh thanks a lot for fixing @cbodley ! I didn't notice rados not being copied at all.